Sage/carbon
# refactor(card): convert to typescript (FE-4774)

### Proposed behaviour

- Refactor `Card` and its subcomponents `CardRow`, `CardFooter` and `CardColumn` to typescript
- Update `Card` documentation to reflect the current version of the component
- Move stories to dedicated stories file written in typescript CSF - to allow for type checking - and import them into docs written in MDX. 
- Refactor draggable Card story so it can be rewritten in typescript. Replace custom colours with design tokens.
- Deprecate `dataRole` prop and replace with `data-role` for consistency with other Carbon components.

https://user-images.githubusercontent.com/18368713/192796546-0f0ce403-9f31-47ee-900a-23122bcf1f52.mov

- Convert test stories file from MDX to typescript CSF

### Current behaviour

- `Card` and its subcomponents are all written in javascript
- All stories are written in MDX - making it harder to flag prop type errors.

### Checklist




#### QA


### UX QA Testing instructions
 
- In storybook, check updated `with draggable` story visuals and the Carbon components it uses (under Show Code option) to verify if story complies with Design System standards.

### QA Testing instructions

- Verify no visual or functional regressions have been introduced from this refactor compared to `master`.
- Verify storybook starts correctly with no console errors.
